samza-core/src/main/scala/org/apache/samza/coordinator/Latch.java

```diff
@@ -30,4 +30,9 @@
 public interface Latch {
   void await(long timeout, TimeUnit tu) throws TimeoutException;
   void countDown();
+
+  /**
+   * implementation specific cleanup
+   */
+  void close();
```

Comment:
Imo, `close` should not belong to the interface api. It is tough to accurately describe the contract the implementation class has to guarantee. In general, if we define the state-changes for the `Latch` interface, then we can reason about whether there is a need for this method to belong to the interface definition. For example, if there was such a thing as `open` for a latch, then, having a corresponding `close` seems reasonable. This is my opinion. 
Will be great if others can chime-in. Thanks!

Reply:
But we need to provide some way for cleanup for all of these utilities. So close() does that.